tests/integration/webclient/src/test/java/io/helidon/tests/integration/webclient/HeadersTest.java

```diff
@@ -0,0 +1,148 @@
+                .get(10, TimeUnit.SECONDS);
+
+        client = WebClient.builder()
+                .baseUri("http://localhost:" + server.port())
+                .validateHeaders(false)
+                .keepAlive(true)
+                .build();
+    }
+
+    @AfterAll
+    static void afterAll() throws ExecutionException, InterruptedException, TimeoutException {
+        if (server != null) {
+            server.shutdown()
+                    .toCompletableFuture()
+                    .get(10, TimeUnit.SECONDS);
+        }
+    }
+
+    // Verify that invalid content type causes an exception when being parsed
+    @Test
+    void testInvalidContentType() {
+        try {
+            client.get()
+                    .path("/invalidContentType")
+                    .request()
+                    .await();
+            fail("WebClient shall throw an exception");
+        } catch (Exception ex) {
+            assertThat(ex, is(instanceOf(CompletionException.class)));
+            Throwable cause = ex.getCause();
+            assertThat(cause, is(notNullValue()));
+            assertThat(cause, is(instanceOf(IllegalArgumentException.class)));
+            assertThat(cause.getMessage(), containsString(INVALID_CONTENT_TYPE_VALUE));
+        }
+    }
+
+    // Verify that "text" content type causes an exception when being parsed in strict mode
+    @Test
+    void testInvalidTextContentTypeStrict() {
+        try {
+        client.get()
+                .path("/invalidTextContentType")
+                .request()
+                .await();
+        } catch (Exception ex) {
+            assertThat(ex, is(instanceOf(CompletionException.class)));
+            Throwable cause = ex.getCause();
+            assertThat(cause, is(notNullValue()));
+            assertThat(cause, is(instanceOf(IllegalArgumentException.class)));
+            assertThat(cause.getMessage(), containsString(INVALID_CONTENT_TYPE_TEXT));
+        }
+    }
+
+    // Verify that "text" content type is transformed to "text/plain" in relaxed mode
+    @Test
+    void testInvalidTextContentTypeRelaxed() {
+        WebClient client = WebClient.builder()
+                .baseUri("http://localhost:" + server.port())
+                .validateHeaders(false)
+                .keepAlive(true)
+                .mediaTypeParserRelaxed(true)
+                .build();
+        WebClientResponse response = client.get()
+                .path("/invalidTextContentType")
+                .request()
+                .await();
+        Optional<MediaType> maybeContentType = response.headers().contentType();
+        assertThat(maybeContentType.isPresent(), is(true));
+        assertThat(maybeContentType.get().toString(), is(RELAXED_CONTENT_TYPE_TEXT));
+    }
+
+    private static final String INVALID_CONTENT_TYPE_VALUE = "invalid header value";
+
+    // HTTP service with invalid Content-Type
+    private static void invalidContentType(ServerRequest request, ServerResponse response) {
```

Comment:
Move also these static methods under the beforeAll method please. I could not find them when trying to figure out what the endpoint does.